File: app/facade/modelmapper/SearchResultsModelMapper.scala
package facade.modelmapper

import domain.{Team, Person}
import model.{PersonSearchResult, TeamSearchResult, DefaultSearchResultsModel}
import scala.collection.JavaConverters._

class SearchResultsModelMapper {
    def map(persons: List[Person], teams :List[Team]): DefaultSearchResultsModel = new DefaultSearchResultsModel(
        persons.map { person => PersonSearchResultMapper.map(person) }.asJava,
        teams.map { team => TeamSearchResultMapper.map(team) }.asJava
    )

    object PersonSearchResultMapper {
        def map(person: Person): PersonSearchResult = new PersonSearchResult(
            person.name,
            person.team.name(),
            person.picture
        )
    }

    object TeamSearchResultMapper {
        def map(team: Team): TeamSearchResult = new TeamSearchResult(team.name)
    }
}

File: app/facade/modelmapper/SearchResultsModelMapper.java
package facade.modelmapper;

import domain.Person;
import domain.Team;
import model.DefaultSearchResultsModel;
import model.PersonSearchResult;
import model.TeamSearchResult;

import java.util.ArrayList;
import java.util.List;

public class SearchResultsModelMapper {
    public DefaultSearchResultsModel map(List<Person> personsList, List<Team> teamsList) {

        List<PersonSearchResult> personSearchResultsList = new ArrayList<>();
        List<TeamSearchResult> teamSearchResultsList = new ArrayList<>();

        for(Person person: personsList) {
            personSearchResultsList.add(PersonSearchResultMapper.map(person));
        }

        for(Team team: teamsList) {
            teamSearchResultsList.add(TeamSearchResultMapper.map(team));
        }

        return new DefaultSearchResultsModel(personSearchResultsList, teamSearchResultsList);
    }

    private static class PersonSearchResultMapper {
        public static PersonSearchResult map(Person person) {
            return new PersonSearchResult(person.name(),person.team().name(), person.picture());
        }
    }

    private static class TeamSearchResultMapper {
        public static TeamSearchResult map(Team team) {
            return new TeamSearchResult(team.name());
        }

    }
}
